vendors/st/boards/stm32l475_discovery/aws_tests/application_code/st_code/stm32l4xx_hal_msp.c

```diff
@@ -1,91 +1,211 @@
+ *    this license is void and will automatically terminate your rights under
+ *    this license.
+ *
+ * THIS SOFTWARE IS PROVIDED BY STMICROELECTRONICS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS, IMPLIED OR STATUTORY WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NON-INFRINGEMENT OF THIRD PARTY INTELLECTUAL PROPERTY
+ * RIGHTS ARE DISCLAIMED TO THE FULLEST EXTENT PERMITTED BY LAW. IN NO EVENT
+ * SHALL STMICROELECTRONICS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ ******************************************************************************
+ */
 
 /* Includes ------------------------------------------------------------------*/
 #include "main.h"
+#define ARD_D13_Pin GPIO_PIN_5
+#define ARD_D13_GPIO_Port GPIOA
+#define ARD_D12_Pin GPIO_PIN_6
+#define ARD_D12_GPIO_Port GPIOA
+#define ARD_D11_Pin GPIO_PIN_7
+#define ARD_D11_GPIO_Port GPIOA
 
 /**
-  * Initializes the Global MSP.
-  */
+ * Initializes the Global MSP.
+ */
 
-void HAL_RNG_MspInit(RNG_HandleTypeDef* hrng)
+void HAL_RNG_MspInit( RNG_HandleTypeDef * hrng )
 {
-  if(hrng->Instance==RNG)
-  {
-      /* Peripheral clock enable */
-    __HAL_RCC_RNG_CLK_ENABLE();
-  }
+    if( hrng->Instance == RNG )
+    {
+        /* Peripheral clock enable */
+        __HAL_RCC_RNG_CLK_ENABLE();
+    }
 }
 
-void HAL_RNG_MspDeInit(RNG_HandleTypeDef* hrng)
+void HAL_RNG_MspDeInit( RNG_HandleTypeDef * hrng )
 {
-  if(hrng->Instance==RNG)
-  {
-    /* Peripheral clock disable */
-    __HAL_RCC_RNG_CLK_DISABLE();
-  }
+    if( hrng->Instance == RNG )
+    {
+        /* Peripheral clock disable */
+        __HAL_RCC_RNG_CLK_DISABLE();
+    }
+}
+
+void HAL_RTC_MspInit( RTC_HandleTypeDef * hrtc )
+{
+    if( hrtc->Instance == RTC )
+    {
+        __HAL_RCC_RTC_ENABLE();
+    }
+}
+
+void HAL_RTC_MspDeInit( RTC_HandleTypeDef * hrtc )
+{
+    if( hrtc->Instance == RTC )
+    {
+        /* Peripheral clock disable */
+        __HAL_RCC_RTC_DISABLE();
+    }
+}
+
+/* Generated by STM32CubeMX. */
+void HAL_UART_MspInit( UART_HandleTypeDef * uartHandle )
+{
+    GPIO_InitTypeDef GPIO_InitStruct = { 0 };
+
+    if( uartHandle->Instance == UART4 )
+    {
+        /* USER CODE BEGIN UART4_MspInit 0 */
+
+        /* USER CODE END UART4_MspInit 0 */
+        /* UART4 clock enable */
+        __HAL_RCC_UART4_CLK_ENABLE();
+
+        __HAL_RCC_GPIOA_CLK_ENABLE();
+        __HAL_RCC_GPIOB_CLK_ENABLE();
+
+        /**UART4 GPIO Configuration
```

Comment:
Are these pin mapping comment blocks all still correct? I'm not convinced that they're easier to understand than the code itself. Perhaps they could be replaced with comments that are purely conversational language.

Reply:
TX and RX are correct but CTS is not. Let me remove those comments.